toolchain/parse/precedence.cpp

```diff
@@ -54,22 +54,21 @@ struct OperatorPriorityTable {
     MarkHigherThan({Highest}, {TermPrefix, LogicalPrefix});
     MarkHigherThan({TermPrefix},
                    {NumericPrefix, BitwisePrefix, IncrementDecrement});
-    MarkHigherThan({NumericPrefix, BitwisePrefix},
+    MarkHigherThan({NumericPrefix, BitwisePrefix, TypePostfix},
                    {As, Multiplicative, Modulo, BitwiseAnd, BitwiseOr,
                     BitwiseXor, BitShift});
     MarkHigherThan({Multiplicative}, {Additive});
     MarkHigherThan(
-        {As, Additive, Modulo, BitwiseAnd, BitwiseOr, BitwiseXor, BitShift},
+        {Additive, Modulo, BitwiseAnd, BitwiseOr, BitwiseXor, BitShift},
         {Relational});
     MarkHigherThan({Relational, LogicalPrefix}, {LogicalAnd, LogicalOr});
-    MarkHigherThan({LogicalAnd, LogicalOr}, {If});
+    MarkHigherThan({As, LogicalAnd, LogicalOr}, {If});
```

Comment:
Based on #4075, I'd expect to see `where` here. Is that just not implemented yet?

Reply:
That is correct, `where` has not yet been added.

Reply:
I wanted to make smaller, incremental PRs, so I left `where` for later.